d3a_interface/client_connections/websocket_connection.py

```diff
@@ -0,0 +1,84 @@
+import asyncio
+import logging
+import threading
+from abc import ABC, abstractmethod
+from time import time
+
+import json
+import traceback
+import websockets
+
+from d3a_interface.client_connections.constants import (
+    WEBSOCKET_ERROR_THRESHOLD_SECONDS,
+    WEBSOCKET_MAX_CONNECTION_RETRIES,
+    WEBSOCKET_WAIT_BEFORE_RETRY_SECONDS)
+from d3a_interface.client_connections.utils import retrieve_jwt_key_from_server
+
+
+class WebsocketMessageReceiver(ABC):
+    def __init__(self, rest_client):
+        self.client = rest_client
+
+    @abstractmethod
+    def received_message(self, message):
+        ...
```

Comment:
Out of curiosity, is this preferred over the old-fashioned `pass` statement ?

Reply:
It looks it is only a matter of preference, there are no rules about this in the PEPs

Reply:
If that's ok for you, I would then use `pass` (as it's used all across our codebase and it's more common I think).

Reply:
@fievelk Yea I'm okay with both options, I'll leave this here though if it's fine